server/memphis_handlers_stations.go

```diff
@@ -1094,177 +1095,129 @@ func (sh StationsHandler) GetPoisonMessageJourney(c *gin.Context) {
+	// 	analytics.SendEvent(user.Username, "user-ack-poison-message")
+	// }
 
 	c.IndentedJSON(200, gin.H{})
 }
 
 func (sh StationsHandler) ResendPoisonMessages(c *gin.Context) {
-	var body models.ResendPoisonMessagesSchema
+	var body models.ResendPoisonMessagesSchemaPg
 	ok := utils.Validate(c, &body, false, nil)
 	if !ok {
 		return
 	}
-	timeout := 500 * time.Millisecond
-	splitId := strings.Split(body.PoisonMessageIds[0], dlsMsgSep)
-	stationName := splitId[0]
-	sn, err := StationNameFromStr(stationName)
-	if err != nil {
-		serv.Errorf("ResendPoisonMessages: " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-		return
-	}
-	streamName := fmt.Sprintf(dlsStreamName, sn.Intern())
-	streamInfo, err := serv.memphisStreamInfo(streamName)
+	err := updatePoisongDlsCg(body.PoisonMessageIds[0], "poison", body.StationName, "ResendPoisonMessages")
 	if err != nil {
 		serv.Errorf("ResendPoisonMessages: " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
 		return
 	}
-	amount := streamInfo.State.Msgs
-	for _, msgId := range body.PoisonMessageIds {
-		filter := GetDlsSubject("poison", sn.Intern(), msgId, "*")
-		msgs, err := serv.memphisGetMessagesByFilter(streamName, filter, 0, amount, timeout)
-		if err != nil {
-			serv.Errorf("ResendPoisonMessages: " + err.Error())
-			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-			return
-		}
-
-		for _, msg := range msgs {
-			var dlsMsg models.DlsMessage
-			err = json.Unmarshal(msg.Data, &dlsMsg)
-			if err != nil {
-				serv.Errorf("ResendPoisonMessages: " + err.Error())
-				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-				return
-			}
-			stationName := replaceDelimiters(dlsMsg.StationName)
-			cgName := replaceDelimiters(dlsMsg.PoisonedCg.CgName)
-			headersJson := map[string]string{}
-			for key, value := range dlsMsg.Message.Headers {
-				headersJson[key] = value
-			}
-			headersJson["$memphis_pm_id"] = dlsMsg.ID
-			headersJson["$memphis_pm_sequence"] = strconv.FormatUint(msg.Sequence, 10)
-			headers, err := json.Marshal(headersJson)
-			if err != nil {
-				serv.Errorf("ResendPoisonMessages: Poisoned consumer group: " + dlsMsg.PoisonedCg.CgName + ": " + err.Error())
-				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-				return
-			}
-			data, err := hex.DecodeString(dlsMsg.Message.Data)
-			if err != nil {
-				serv.Errorf("ResendPoisonMessages: Poisoned consumer group: " + dlsMsg.PoisonedCg.CgName + ": " + err.Error())
-				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-				return
-			}
-			err = sh.S.ResendPoisonMessage("$memphis_dls_"+stationName+"_"+cgName, []byte(data), headers)
-			if err != nil {
-				serv.Errorf("ResendPoisonMessages: Poisoned consumer group: " + dlsMsg.PoisonedCg.CgName + ": " + err.Error())
-				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-				return
-			}
-		}
-
-	}
-
-	shouldSendAnalytics, _ := shouldSendAnalytics()
```

Comment:
You should those lines. please don't delete line that relates to shouldSendAnalytics()